bigchaindb/models.py

```diff
@@ -101,11 +73,28 @@ def validate(self, bigchain):
 
                 input_conditions.append(input_tx.conditions[input_cid])
                 input_txs.append(input_tx)
+                input_amount += input_tx.conditions[input_cid].amount
 
             # validate asset id
             asset_id = Asset.get_asset_id(input_txs)
             if asset_id != self.asset.data_id:
-                raise AssetIdMismatch('The asset id of the input does not match the asset id of the transaction')
+                raise AssetIdMismatch(('The asset id of the input does not'
+                                       ' match the asset id of the'
+                                       ' transaction'))
+
+            # get the asset creation to see if its divisible or not
+            asset = bigchain.get_asset_by_id(asset_id)
+            # validate the asset
+            asset._validate_asset(amount=input_amount)
+            # validate the amounts
+            output_amount = sum([condition.amount for
+                                 condition in self.conditions])
+            if output_amount != input_amount:
+                raise AmountError(('The amout used in the inputs `{}`'
```

Comment:
There's a minor typo here: `The amout used` should be `The amount used`.

Maybe a more helpful message would be something like 'The sum of the input amounts (14) differs from the sum of the output amounts (23)'